e2e-tests/native/test/utils/deviceUtils.ts

```diff
@@ -1,5 +1,11 @@
 import { Selector } from '../helpers/Selector.js'
 
 export const allowNotifications = async (): Promise<void> => {
-  await $(new Selector().ByText('Allow').build()).click()
+  try {
+    const allow = await $(new Selector().ByText('Allow').build())
+    await allow.waitForDisplayed()
+    allow.click()
+  } catch (error) {
+    console.warn('Permission request was not displayed within the timeout:', error)
+  }
```

Comment:
💭  This looks to me like it would mean that the tests would pass even if the request for permissions never show up. If that is the case, and if that is what we want to do, we can just revert #2761

Reply:
Yes you are right. I think if we want to check if the permission dialog is displayed we should just add a own test for this, but we do E2E tests only for really a few very basic use cases, so I think it is not needed.

Also reverting the previous PR does not really make sense, as we want to click the "apply" button and proceed.

Reply:
I'm okay with adding a separate test for whether the permission is requested but I'm not comfortable with leaving the test like this because it doesn't actually fail if the permission isn't requested. Instead, we have a test that looks like it tests for the permissions but doesn't, leading to a false sense of security, and a completely useless test.

Reply:
I am confused now. We have four tests "changeLanguage", "navigateToLeafCategory", "navigateToOtherLocation" and "navigateToSearchResults". Neither of them does have a test for requesting permissions, it is only one step in the "should open a city on location selection" test. So I would not expect this test to check for correct permission request handling.

Reply:
You're right, we don't actually have a separate test for the notifications permission. I'll open an issue for it.